oop1-construct.js

```diff
@@ -1,16 +1,62 @@
+//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.  This string should clearly explain why classes are used in Object Oriented Programming. 
+// The first method is called explain().  This method should print the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that are essential to building a class.  
+//Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
 
 //your code here...
+class WhyClass {
+    constructor(purpose) {
+
+        this.purpose = purpose;
+    }
+    explain(purpose) {
+        console.log(purpose);
+    }
+    pieces() {
+        console.log('The important pieces of a class are the contructor, getters, setters, methods and attributes.')
+    }
+
 
 
+}
+
+let example = new WhyClass("a class is a blueprint for creating objects (a particular data structure), providing initial values for state (member variables or attributes), and implementations of behavior (member functions or methods). The user-defined objects are created using the class keyword.");
+
+example.explain(example.purpose);
+example.pieces();
 
-/*2. You are exploring the rainforests of the Amazon.  You have observed many different types of wildlife on your excursion.  To jog your memory, some different types of wildlife that you observed are found in the following URL : https://www.ietravel.com/blog/amazon-rainforest-animals-beginners-guide-21-species.  Feel free to use other resources online if you wish.
 
-Create a class that can categorize at least 6 of the creatures that you recall observing the Amazon.  Each creature should have 3 unique characteristics, and 2 unique actions that it can do. */
+/*2. You are exploring the rainforests of the Amazon.  You have observed many different types of wildlife on your excursion.  
+//To jog your memory, some different types of wildlife that you observed are found in the following URL :
+// https://www.ietravel.com/blog/amazon-rainforest-animals-beginners-guide-21-species.  
+//Feel free to use other resources online if you wish.
+
+Create a class that can categorize at least 6 of the creatures that you recall observing the Amazon.  
+//Each creature should have 3 unique characteristics, and 2 unique actions that it can do. */
 
 
 //your code here...
+class Mammal {
+    constructor(name, color, food, transit, noise) {
+        name;
+        color;
+        food;
+        transit;
+        noise;
+    }
```

Comment:
You have to assign the values you pass into the constructor, or they're lost.
` constructor(name, color, food, transit, noise) {
        this.name = name;
        this.color = color;
        this.food = food;
        this.transit = transit;
        this.noise = noise;
    }`